### PR TITLE
add Dockerfile

### PR DESCRIPTION
**Pull request by lazysoundsystem**
As mentioned in #78, here's a dockerfile. 

There's a bit of a mess around including reports.json, some of which can be cleaned up when the npm package is updated.

### CODE REVIEW

File: Dockerfile.example [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# Dockerfile for analytics-reporter.
+# https://github.com/18F/analytics-reporter
+#
+# For information about setting up keys, etc, see README.md.
+# This Dockerfile requires keys, and optional customizations, to be set below.
+##
+
+FROM debian:jessie
+RUN apt-get update && apt-get install -yq \
+  nodejs-legacy \
```
Comment (konklone) [inline Dockerfile.example]: Sigh. I wonder if we can use `iojs`? Is that even in apt?

File: Dockerfile.example [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+# Add pem key. Requires the pem key to be named secret_key.pem and present
+# in the same directory.
+##
+RUN mkdir /opt/analytics
+COPY ./secret_key.pem /opt/analytics/secret_key.pem
+ENV ANALYTICS_KEY_PATH="/opt/analytics/secret_key.pem"
+##
+# These two must be changed!
+##
+ENV ANALYTICS_REPORT_IDS="ga:XXXXXX"
```
Comment (konklone) [inline Dockerfile.example]: Can't these be stored outside of the Dockerfile...? Feels like it's asking for trouble for people to add private information to a versioned file.

**Comment by konklone** [comment]
I'm happy to do an npm push whenever, in coordination with @laurenancona to pick our next version number.

**Comment by lazysoundsystem** [comment]
This builds directly off an iojs container, and, I think, handles the environment variables a bit better.

**Comment by konklone** [comment]
This is *awesome*, thank you @lazysoundsystem! I'm happy to merge, but I did just realize that we currently have documented support for `.env` files, using `autoenv`, which might fulfill the role of `config.txt` already.

There's an [`env.example`](https://github.com/18F/analytics-reporter/blob/master/env.example) file in the repo already, which the documentation encourages people to copy to `.env` and fill in. Easy to change `config.txt` to `.env`? (Removes the need for a line in `.gitignore` too.)

**Comment by konklone** [comment]
I'm trying to verify this container, and I keep getting `invalid_grant` errors when running it, even though if I shell into the container with `/bin/bash`, I can run `analytics --only devices` and get output.

I did have to change my local `.env` to point the key path to the Docker-space path to get this far:

```
export ANALYTICS_KEY_PATH=/opt/analytics/secret_key.pem
```

Have you definitely run this Dockerfile and gotten data? Can you share your commands?

**Comment by lazysoundsystem** [comment]
Thanks again @konklone for quick and very useful feedback. 

I moved the variables to .env, which makes sense, but discovered the problem with setting the variables through .bashrc - they only work when run through /bin/bash. Like you say, entering the container with /bin/bash and running analytics works, but it doesn't work from outside the container.

It looks like the two options are:
Setting the variables with ENV in the Dockerfile and keeping it out of version control (include a Dockerfile.example and add 'Dockerfile' to the .gitignore). 
Docker-compose now has an env-file option, I'll look into this.

**Comment by konklone** [comment]
> Docker-compose now has an env-file option, I'll look into this.

This would be my preferred method, so that setting values in `.env` does the job both for straight local development _and_ for Docker-based execution.

Thanks for tackling this! This is getting me back into the Docker saddle and makes me want to do this on a couple of other 18F projects.

**Comment by konklone** [comment]
Just a note that I'm still very happy to help test/review and get this merged. :+1:

**Comment by konklone** [comment]
I hate doing this, but it looks like this has stalled. Happy to re-open if we can resolve my question in #103.

**Pull request closed without merging**